src/main/scala/io/iohk/ethereum/blockchain/sync/Blacklist.scala

```diff
@@ -0,0 +1,161 @@
+package io.iohk.ethereum.blockchain.sync
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import com.github.blemale.scaffeine.{Cache, Scaffeine}
+import io.iohk.ethereum.utils.Logger
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters._
+import scala.jdk.DurationConverters._
+
+import Blacklist._
+import io.iohk.ethereum.network.PeerId
+import io.iohk.ethereum.blockchain.sync.Blacklist.BlacklistReason.BlacklistReasonType.WrongBlockHeadersType
+import io.iohk.ethereum.consensus.validators.std.StdBlockValidator.BlockError
+import io.iohk.ethereum.blockchain.sync.Blacklist.BlacklistReason.BlacklistReasonType
+
+trait Blacklist {
+  def isBlacklisted(id: BlacklistId): Boolean
+  def add(id: BlacklistId, duration: FiniteDuration, reason: BlacklistReason): Unit
+  def remove(id: BlacklistId): Unit
+  def keys: Set[BlacklistId]
+}
+
+object Blacklist {
+  import BlacklistReason._
+  import BlacklistReasonType._
+
+  trait BlacklistId {
+    def value: String
+  }
+
+  sealed trait BlacklistReason {
```

Comment:
sealed trait BlacklistReason {
           def reasonType: BlacklistReasonType
           }
           
sealed trait BlacklistReasonType {
      def name: String
    }
    
    -> so then case objects of BlacklistReasonType are left almost the same, but without codes (unless there's a usage for them)
    ->  all BlacklistReasons are turned to case objects, description is discarded.
    
    
    PS: just a suggestion for simplification, if you think it makes more sense to keep it like it is - I'll approve it like that

Reply:
Thanks for the suggestion but I disagree about removing information that we currently log and that could be helpful for debugging.